overlord/overlord.go

```diff
@@ -102,6 +103,7 @@ func New() (*Overlord, error) {
 	o.hookMgr = hookMgr
 	o.stateEng.AddManager(o.hookMgr)
 
+	hooks.SetupHooks(o.hookMgr)
```

Comment:
Feels like a curious place to do this. The Register method is inside the hookMgr itself, so why would we expose SetupHooks just so something else can call into the package for it to register its own hooks on its own type?  Can't we do this inside the Manager function itself, and not export the function?